generate_markdown_from_doxygen_xml.py

```diff
@@ -0,0 +1,1012 @@
+            for the_type in self.public_typedef:
+                outputstring+=the_type.markdown()
+
+        if len(self.public_enums)>0:
+            outputstring+='\n\n## Member Enumeration Documentation\n\n'
+            for the_type in self.public_enums:
+                outputstring+=the_type.markdown()
+
+        ## Member Function Documentation
+        if len(self.public_functions)>0 or len(self.public_static_functions)>0:
+            outputstring+='\n\n## Member Function Documentation\n\n'
+            if len(self.public_functions)>0:
+                #print(self.public_functions)
+                for the_function in self.public_functions:
+                    outputstring+=the_function.markdown()
+
+            if len(self.public_static_functions)>0:
+                #print(self.public_static_functions)
+                for the_function in self.public_static_functions:
+                    outputstring+=the_function.markdown()
+        
+        ## Protected Member Functions
+        if len(self.protected_functions)>0:
+            outputstring+='\n\n## Protected Member Functions\n'
+            for the_function in self.protected_functions:
+                outputstring+=the_function.markdown()
+                
+        ## Member Data Documentation 
+        if len(self.protected_attribs)>0:
+            outputstring+='\n\n## Member Data Documentation\n\n'
+            for the_item in self.protected_attribs:
+                outputstring+=the_item.markdown()
+                
+        ## Field Documentation
+        if len(self.public_static_attributes)>0 or len(self.public_attributes)>0:
+            outputstring+='\n\n## Field Documentation\n\n'
+            if len(self.public_static_attributes)>0:
+                for the_item in self.public_static_attributes:
+                    outputstring+=the_item.markdown()
+                    
+            if len(self.public_attributes)>0:
+                for the_item in self.public_attributes:
+                    outputstring+=the_item.markdown()
+                
+            
+            
+        outputstring=cleanup_markdown_string(outputstring)
+            
+        return outputstring
+
+
+# Start actual execution
+parser = argparse.ArgumentParser(description='Generate markdown from doxygen XML output.')
+parser.add_argument("-d", "--debug", default='',help="Any argument used here will include debug info in markdown output - ie some strings for various objects.")
+args = parser.parse_args()
+
+
+
+# Check if XML file path exists
+if not os.path.exists(DOXYGEN_ROOT_DIR):
+    print('Docs directory does not exist. Run: `make docs`')
+    exit()
+if not os.path.exists(DOXYGEN_XML_DIR):
+    print('XML directory does not exist (created by doxygen). Run: `make docs`')
+    exit()
+if not os.path.exists(DOXYGEN_OUTPUT_DIR):
+    #make output dir
+    os.makedirs(DOXYGEN_OUTPUT_DIR)
+
+
+
+for root, dirs, files in os.walk(DOXYGEN_XML_DIR, topdown=False):
```

Comment:
@julianoes See, here I walk the XML folder. Ignores lots of file types that are really "internal" - may not be coming out of this doxygen file, but were in my test file.